test-data/unit/check-based-infer-function-types.test

```diff
@@ -203,6 +203,32 @@ def f(**kwargs):  # E: Function is missing a type annotation for one or more arg
 [builtins fixtures/dict.pyi]
 
 
+[case testInferOverloadWithPositionalOnly]
+from typing import overload
+
+@overload
+def foo(a: int, /): ...
+@overload
+def foo(a: str, /): ...
+
+def foo(a):
+    reveal_type(a)
+
+@overload
+def bar(x: int, /, b: str): ...
+@overload
+def bar(y: None, /, b: ellipsis): ...
+
+def bar(z, b):
+    reveal_type(z)
+    reveal_type(b)
+[out]
+main:4: error: invalid syntax  [syntax]
+[out version>=3.8]
+main:9: note: Revealed type is "int | str"
+main:17: note: Revealed type is "int | None"
+main:18: note: Revealed type is "str | ellipsis"
+
```

Comment:
add tests for `*` as well? and a test for if one overload has a `/` and the other doesnt? etc

Reply:
done?

Reply:
hmm, what happens if the `/` or `*` is in a different position in each overload

Reply:
added

Reply:
did u push? i don't see it

Reply:
https://github.com/KotlinIsland/basedmypy/compare/753aa264c8dd4316c5d7d20f9184e95f5bd57a2d..95acb2286b1b4506dfdabbb2b49e38b07cc0b4b8

Reply:
or did you mean something else?

Reply:
thats a good one too, tho i meant something like
```py
@overload
def f(x: int, /, b: int): ...
@overload
def f(x: int, b: int, /): ...
```

Reply:
sure

Reply:
nah